Bitbucket.Authentication/Test/AuthenticationTest.cs

```diff
@@ -473,6 +473,46 @@ public async void VerifyInteractiveLoginDoesNotAquireInvalidBasicAuthCredentials
             credentialStore.Verify(c => c.WriteCredentials(It.IsAny<TargetUri>(), It.IsAny<Credential>()), Times.Never);
         }
 
+        [Fact]
+        public async void VerifyInteractiveLoginDoesNotAquireInvalidBasicAuthCredentialsWithUsername()
+        {
+            var bitbucketUrl = "https://bitbucket.org";
+            var credentialStore = new Mock<ICredentialStore>();
+
+            // mock the result that normally causes issues
+            var validAuthenticationResult = new AuthenticationResult(AuthenticationResultType.Success)
+            {
+                Token = new Token(_validPassword, TokenType.Personal),
+                RemoteUsername = _validUsername
+            };
+
+            var targetUri = new TargetUri(bitbucketUrl);
+
+            // Mock the behaviour of IAuthority.AcquireToken() to basically mimic BasicAuthAuthenticator.GetAuthAsync() validating the useername/password
+            var authority = new Mock<IAuthority>();
+            authority
+                .Setup(a => a.AcquireToken(It.IsAny<TargetUri>(), It.IsAny<Credential>(), It.IsAny<AuthenticationResultType>(), It.IsAny<TokenScope>()))
+                // return 'success' with the validated credentials
+                .Returns(Task.FromResult(validAuthenticationResult));
+
+            var bbAuth = new Authentication(RuntimeContext.Default, credentialStore.Object, 
+                MockInvalidBasicAuthCredentialsAquireCredentialsCallback, MockValidAquireAuthenticationOAuthCallback, authority.Object);
+
+            // perform login with username
+            var credentials = await bbAuth.InteractiveLogon(targetUri, _validUsername);
+
+            Assert.NotNull(credentials);
+            Assert.Equal(_validUsername, credentials.Username);
```

Comment:
The critical part of the test -- make sure we get back a username that's valid and not "Personal Access Token". We didn't previously have a test for if `AuthenticationResultType.Success` which is why this didn't get caught in the past.